### PR TITLE
More changes for Tanssi Data Preservers container chain spawning on assignment

### PR DESCRIPTION
**Pull request by nanocryk**
Companion of https://github.com/moondance-labs/tanssi/pull/625
- Adds new functions to `OrchestratorChainInterface` to get rid of the need for a client+runtime api
- Remove functions added in previous PR as they are only used for collation, which will now still have access to the client.
-  Move container chain primitive to dancekit (to be used by the interface functions)
- Remove RPC impl of  `OrchestratorChainInterface`  which will be moved to Tanssi

### CODE REVIEW

File: primitives/container-chain-genesis-data/src/json.rs [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    Ok(properties)
+}
+
+#[cfg(test)]
+mod tests {
+    use sp_core::ConstU32;
+
+    use super::*;
+
+    fn expected_container_chain_genesis_data() -> ContainerChainGenesisData<ConstU32<255>> {
```
Comment (tmpolaczyk) [inline primitives/container-chain-genesis-data/src/json.rs:221]:
```suggestion
    fn expected_container_chain_genesis_data() -> ContainerChainGenesisData {
```

Why is the CI green when this shouldn't compile?

Comment (nanocryk) [reply primitives/container-chain-genesis-data/src/json.rs:221]: Feature-gated `json`. Since locally tanssi compiles it seems this feature is not used?

Comment (tmpolaczyk) [reply primitives/container-chain-genesis-data/src/json.rs]: Probably that feature is enabled for build but disabled when running tests...

Comment (tmpolaczyk) [reply primitives/container-chain-genesis-data/src/json.rs]: I checked the CI script and we do the same as in the tanssi repo, so we will encounter this issue there as well, as long as there is some feature which is not enabled by default, our unit tests will not run :confused: 

I guess the easiest way to notice that is to check the generated coverage file

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**